### DISCUSSION TITLE
Radio Buttons

### DISCUSSION THREAD

**Discussion by nonnorm** [slint-ui/slint#2077] [Q&A] [answered]
I've seen radio buttons mentioned vaguely in the docs (I think) and in PRs, but are there actually radio buttons built in to slint? Otherwise, is there a way to make checkboxes behave like radio buttons?

**Comment by ogoffart** [comment, marked as answer]
We still don't have builtin radio button.

We have an example that implement them:
https://github.com/slint-ui/slint/blob/29b17fd10977a3da6c01c8a9df27b5ba642d06a7/examples/fancy_demo/main.slint#L197
Used in 
https://github.com/slint-ui/slint/blob/29b17fd10977a3da6c01c8a9df27b5ba642d06a7/examples/fancy_demo/main.slint#L387-L389
and we manually reset them on click with the function https://github.com/slint-ui/slint/blob/29b17fd10977a3da6c01c8a9df27b5ba642d06a7/examples/fancy_demo/main.slint#L358

But this is a currently missing widget that we still need to add to Slint

**Reply by codecat** [reply]
Is there a way to do this when the number of radio buttons are variable? (eg. generated dynamically from an array?)

Edit: Figured out a way, posted as a new reply.

**Comment by codecat** [comment]
In addition to the answer already in this thread, if you have a dynamic number of radio buttons, you can do something like this:
```
VerticalLayout {
  out property <string> selected-item;

  for item in items : RadioButton {
    text: item;
    checked: selected-item == item;
    clicked => { selected-item = item; }
  }
}
```
